Unity/Assets/Game/Audio/AudioManager.cs

```diff
@@ -0,0 +1,195 @@
+		}
+
+		AudioSource sound = _audiosources[soundName];
+		sound.outputAudioMixerGroup = _sfxMixer;
+		if (!sound.isPlaying) sound.Play();
+		return sound;
+	}
+	
+	/// <summary>
+	/// Plays the music specified on loop. Crossfades between last played music and the new one.
+	/// Creates an audiosource if it doesn't already exist.
+	/// </summary>
+	/// <param name="musicName">Name of the music to be played, case sensitive.</param>
+	/// <param name="crossfadeTime">Time for crossfade to occur. Negative value denotes that it will use the
+	/// default crossfade time specified in the audiosettings SO.</param>
+	public void SetMusic(string musicName, float crossfadeTime = -1)
+	{
+		if (musicName == _currentMusic || _crossfading) return;
+		if (crossfadeTime < 0) crossfadeTime = _audioSettings.crossfadeTime;
+		
+		if (!_audiosources.ContainsKey(musicName))
+		{
+			_audiosources.Add(musicName, gameObject.AddComponent<AudioSource>());
+			_audiosources[musicName].clip = _audioSettings.bgmDictionary[musicName];
+		}
+
+		AudioSource newMusic = _audiosources[musicName];
+		newMusic.Play();
+		newMusic.loop = true;
+		newMusic.outputAudioMixerGroup = _bgmMixer;
+
+		_crossfading = true;
+		LeanTween.value(0.0f, 1.0f, crossfadeTime)
+			.setOnUpdate((value) =>
+			{		
+				if (_currentMusic != "")
+					_audiosources[_currentMusic].volume = 1 - value;
+				
+				_audiosources[musicName].volume = 1;
+			})
+			.setOnComplete(() =>
+			{
+				_currentMusic = musicName;
+				_crossfading = false;
+			});
+	}
+
+	#endregion
+
+	#region Set Volume Functions
+
+	/// <summary>
+	/// Sets the master volume, which attenuates both Sound Effects (SFX) and background music (BGM).
+	/// </summary>
+	/// <param name="volume">Should be between 0 (muted) and 1 (full volume)</param>
+	public void SetVolumeMaster(float volume)
+	{
+		SetVolume(volume, "VolumeMaster");
+	}
+	
+	/// <summary>
+	/// Sets the volume for Sound Effects (SFX).
+	/// </summary>
+	/// <param name="volume">Should be between 0 (muted) and 1 (full volume)</param>
+	public void SetVolumeSFX(float volume)
+	{
+		SetVolume(volume, "VolumeSFX");
+	}
+
+	/// <summary>
+	/// Sets the volume for background music (BGM).
+	/// </summary>
+	/// <param name="volume">Should be between 0 (muted) and 1 (full volume)</param>
+	public void SetVolumeBGM(float volume)
+	{
+		SetVolume(volume, "VolumeBGM");
+	}
+
+	/// <summary>
+	/// Helper function for setting exposed volume parameters of _masterMixer
+	/// </summary>
+	/// <param name="volume">Should be between 0 (muted) and 1 (full volume)</param>
+	/// <param name="parameterName">Name of exposed parameter, must match exactly, case sensitive</param>
+	private void SetVolume(float volume, string parameterName)
+	{
+		volume = Mathf.Clamp(volume, 0.0001f, 1.0f);
+		volume =  Mathf.Log10(volume) * 20; //-80db is muted, 0 is regular "full" volume
+		//TODO set to logarithmic scaling
+		_masterMixer.SetFloat(parameterName, volume); 
```

Comment:
Was this TODO something you've already done or something you want to do? Please clean up this type of comment in the future